hugo/assets/js/species_cards.js

```diff
@@ -0,0 +1,284 @@
+    const dropdown = document.querySelector(".scilife-sort-dropdown");
+    const dropdownButtons = Object.values(SORT_OPTIONS);
+    dropdownButtons.forEach(id => document.getElementById(id).classList.remove("active"));
+
+    const activeButton = document.getElementById(state.sortOrder);
+    activeButton.classList.add("active");
+
+    const templateId = `text-${state.sortOrder}`;
+    const template = document.getElementById(templateId);
+    const clone = template.content.cloneNode(true);
+    dropdown.innerHTML = '';
+    dropdown.appendChild(clone);
+}
+
+
+/**
+ * Updates the page number shown as active in the pagination
+ */
+function updateActivePage() {
+    paginationItems.forEach(item => {
+        item.classList.remove('active');
+    });
+    const activePage = document.querySelector(`.pagination .page-link[data-page="${state.currentPage}"]`).parentElement;
+    activePage.classList.add('active');
+}
+
+/**
+ * Update the page number based on on the provided page identifier.
+ *
+ * @param {string|number} pageSelection - Can be 'prev', 'next', or a specific page number.
+ */
+function changeCurrentPage(pageSelection) {
+    if (pageSelection === 'prev') {
+        state.currentPage = Math.max(state.currentPage - 1, 1);
+    } else if (pageSelection === 'next') {
+        state.currentPage = Math.min(state.currentPage + 1, Math.ceil(state.speciesData.length / CARDS_PER_PAGE));
+    } else {
+        state.currentPage = parseInt(pageSelection);
+    }
+    updateActivePage();
+}
+
+/**
+ * Sets which buttons are enabled/disabled based on the number of pages with results and the current page.
+ */
+function updatePaginationButtons() {
+    const numbActivePages = Math.ceil(state.numbMatches / CARDS_PER_PAGE)
+
+    paginationItems.forEach((item) => {
+        const pageLink = item.querySelector('.page-link');
+        const page = pageLink.getAttribute('data-page');
+        if (page === 'prev') {
+            item.classList.toggle('disabled', state.currentPage === 1);
+        } else if (page === 'next') {
+            item.classList.toggle('disabled', state.currentPage >= numbActivePages);
+        } else {
+            const pageIndex = parseInt(page);
+            item.classList.toggle('disabled', pageIndex > numbActivePages);
+        }
+    });
+}
+
+
+/**
+ * Debounce a function (used for species search)
+ * @param {Function} callback - Function to debounce.
+ * @param {number} delay - Delay in ms.
+ * @returns {Function} - A debounced version of the function.
+ */
+function debounce(callback, delay) {
+    let timer
+    return function (...args) {
+        clearTimeout(timer)
+        timer = setTimeout(() => {
+            callback.apply(this, args);
+        }, delay)
+    }
+}
+
+/**
+ * Handles the search input event.
+ *
+ * @param {Event} event - The input event from the search box.
+ */
+function speciesSearch(event) {
+    state.searchText = event.target.value.toLowerCase();
+    if (PAGINATION_EXISTS) {
```

Comment:
Consider:
```javascript
state.searchText = event.target.value.toLowerCase();
state.currentPage = 1
species = filterSpecies({...state})
page = getCurrentPage(species, {...state})
renderSpeciesCards(page, root)
```

Reply:
I've now implemented something much closer to this! (see main comment)